hibernate-core/src/main/java/org/hibernate/LockModeException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * Copyright (c) 2008, 2013, Red Hat Inc. or third-party contributors as
+ * indicated by the @author tags or express copyright attribution
+ * statements applied by the authors.  All third-party contributions are
+ * distributed under license by Red Hat Inc.
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.hibernate;
+
+/**
+ * This exception is thrown when an invalid LockMode is selected for an entity. This occurs if the
+ * user tries to set an inappropriate LockMode for an entity.
+ *
+ * @author John O'Hara
+ */
+public class LockModeException extends HibernateException {
```

Comment:
Very much dislike this name given its current use.  There are other cases in the codebase where an "incorrect lock mode" might be requested, but you do not apply this in those cases.  Also, it has nothing to do with the "appropriateness" of the LockMode at all in your usage.

So say what it does.  Really this is an exception indicating that an immutable entity cannot be locked.  Or slightly more broadly, an illegal attempt to lock an immutable (unloackable) entity.  So IllegalLockAttemptException?  Or UnsupportedLockAttemptException?  I think I personally prefer the second form.